crates/turbo-tasks/src/task/task_input.rs

```diff
@@ -335,7 +348,7 @@ macro_rules! tuple_impls {
                         )+
                         Ok(($($name,)+))
                     }
-                    _ => bail!("invalid task input type, expected list"),
+                    _ => bail!("invalid task input type {input:?}, expected tuple {}", stringify!(($($name,)+))),
```

Comment:
I expect this function to be heavily monomorphized. It's a minor optimization, but it might be worthwhile to move absolutely everything that does not depend on the type parameters to helper functions, namely `iter.next().ok_or_else(|| anyhow!("missing tuple element"))?` and the `bail!` macro.

This is what I was doing with `get_args`/`get_args_iter`/`next_arg` in `TaskFnInputFunction`: https://github.com/vercel/turbo/pull/8304

Reply:
You are right, but this will be removed in the following PR anyway